WikiSentiment/WikiData/DailyCollection.cs

```diff
@@ -0,0 +1,135 @@
+                    regionalCollections[iCountry] = result; }
+                catch (Exception e) {
+                    logger.LogError($"Error building {iCountry}-collection for " +
+                        $"{_date.Year}-{_date.Month:D2}-{_date.Day:D2}: {e.Message}"); }
+            }
+
+            var featuredCountries = getFeaturedCountries(regionalCollections, featureArticles);
+
+            return new DailyCollection()
+            {
+                countrydailydict = regionalCollections,
+                featuredlist = featuredCountries
+            }; ;
+        }
+
+        /// <summary>
+        /// Updates old colection with new content, compiles new featured list
+        /// </summary>
+        /// <param name="_base"></param>
+        /// <param name="_newAdditions"></param>
+        /// <returns></returns>
+        public static DailyCollection UpdateGiven(DailyCollection _base, DailyCollection _newAdditions)
+        {
+            var newCollection = new Dictionary<string, LanguageCollection>(_base.countrydailydict);
+
+            //override old data with new ones
+            foreach(string iCountry in _newAdditions.countrydailydict.Keys)
+            {
+                newCollection[iCountry] = _newAdditions.countrydailydict[iCountry];
+            }
+            var featured = getFeaturedCountries(newCollection, featureArticles);
+
+            return new DailyCollection()
+            {
+                countrydailydict = newCollection,
+                featuredlist = featured
+            };
+        }
+
+        /// <summary>
+        /// Get ordered list of languages with the top viewed (proportionally) articles
+        /// </summary>
+        /// <param name="_collection"></param>
+        /// <param name="_featuredAmount">amount of languages to keep</param>
+        /// <returns>list of two letter language codes</returns>
+        static List<string> getFeaturedCountries(
+            Dictionary<string, LanguageCollection> _collection, int _featuredAmount)
+        {
+            //compile the list of country codes and viewership scores for top articles
+            var languageTopArticlesData = new List<(string countrycode, float percentage)>();
+            foreach (string iLanguage in _collection.Keys)
+            {
+                languageTopArticlesData.Add((iLanguage,
+                    (100f * _collection[iLanguage].articles[0].vws) / _collection[iLanguage].totalviews));
+            }
+
+            //order data in the descending order
+            languageTopArticlesData = (languageTopArticlesData.OrderBy(data => data.percentage)).Reverse().ToList();
+
+            var amountToFeature = Math.Min(_featuredAmount, _collection.Count);
+            List<string> result = new List<string>();
+
+            //return countrycodes
+            for (int i = 0; i < amountToFeature; i++)
+                result.Add(languageTopArticlesData[i].countrycode);
+            
+            return result;
```

Comment:
Another LINQ, this time [`Take(int count)`](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.take?view=net-8.0)
```suggestion
            return languageTopArticlesData
                .Take(_featuredAmount)
                .Select(x => x.countryCode)
                .ToList();
```